site/app/controllers/forum/ForumController.php

```diff
@@ -172,6 +189,67 @@ public function addNewCategory(){
         return $result;
     }
 
+    public function deleteCategory(){
+        $result = array();
+        if($this->core->getUser()->getGroup() <= 2){
+            if(!empty($_REQUEST["deleteCategory"])) {
+                $category = (int)$_REQUEST["deleteCategory"];
+                if(!$this->isValidCategories(array($category))) {
+                    $result["error"] = "That category doesn't exists.";
+                } else if(!$this->isCategoryDeletionGood($category)) {
+                    $result["error"] = "Last category can't be deleted.";
+                } else {
+                    if($this->core->getQueries()->deleteCategory($category)) {
+                        $result["success"] = "OK";
+                    } else {
+                        $result["error"] = "Category is in use.";
+                    }
+                }
+            } else {
+                $result["error"] = "No category data submitted. Please try again.";
+            }
+        } else {
+            $result["error"] = "You do not have permissions to do that.";
+        }
+        $this->core->getOutput()->renderJson($result);
+        return $result;
+    }
+
+    public function editCategory(){
+        $result = array();
+        if($this->core->getUser()->getGroup() <= 2){
+            $category_id = $_REQUEST["category_id"];
+            $category_desc = null;
+            $category_color = null;
+            $should_update = true;
+
+            if((!empty($_REQUEST["category_desc"])) && isset($_REQUEST["category_desc"])) {
```

Comment:
Take a look at [https://www.codepunker.com/blog/is-null-vs-empty-vs-isset-one-lesson-all-php-coders-should-learn](https://www.codepunker.com/blog/is-null-vs-empty-vs-isset-one-lesson-all-php-coders-should-learn)

Also refer to [http://php.net/manual/en/function.isset.php](http://php.net/manual/en/function.isset.php) and [http://php.net/manual/en/function.empty.php](http://php.net/manual/en/function.empty.php)

Reply:
Thanks! I didn't know that.